src/main/java/com/minecolonies/coremod/colony/buildings/views/MobEntryView.java

```diff
@@ -0,0 +1,183 @@
+        buf.writeBoolean(entry.getAttack());
+        buf.writeInt(entry.getPriority());
+    }
+
+    /**
+     * Reads the Location, Attack, and Priority from a {@link ByteBuf} to create a MobEntryView
+     *
+     * @param buf Buf to read from.
+     * @return MobEntryView that was created.
+     */
+    @NotNull
+    public static MobEntryView readFromByteBuf(@NotNull final ByteBuf buf)
+    {
+        final ResourceLocation location = new ResourceLocation(ByteBufUtils.readUTF8String(buf));
+        final Boolean attack = buf.readBoolean();
+        final Integer priority = buf.readInt();
+
+        return new MobEntryView(location, attack, priority);
+    }
+
+    /**
+     * Writes the Location, Attack, and Priority to NBT
+     *
+     * @param compound Compound to write to.
+     * @param name     Name of the tag.
+     * @param entry      the View to write
+     */
+    public static void writeToNBT(@NotNull final NBTTagCompound compound, final String name, @NotNull final MobEntryView entry)
+    {
+        @NotNull final NBTTagCompound coordsCompound = new NBTTagCompound();
+        coordsCompound.setString("location", entry.getLocation().toString());
+        coordsCompound.setBoolean("attack", entry.getAttack());
+        coordsCompound.setInteger("priority", entry.getPriority());
+        compound.setTag(name, coordsCompound);
+    }
+
+    /**
+     * Reads the Location, Attack, and Priority from nbt to create a MobEntryView
+     *
+     * @param compound Compound to read data from.
+     * @param name     Tag name to read data from.
+     * @return The new MobEntryView
+     */
+    @NotNull
+    public static MobEntryView readFromNBT(@NotNull final NBTTagCompound compound, final String name)
+    {
+        final NBTTagCompound entryCompound = compound.getCompoundTag(name);
+        final ResourceLocation location = new ResourceLocation(entryCompound.getString("location"));
+        final Boolean attack = entryCompound.getBoolean("attack");
+        final Integer priority = entryCompound.getInteger("priority");
+        return new MobEntryView(location, attack, priority);
+    }
+
+    /**
+     * Priority getter.
+     *
+     * @return the priority.
+     */
+    public int getPriority()
+    {
+        return priority;
+    }
+
+    /**
+     * Setter for the priority.
+     *
+     * @param priority the new priority attribute.
+     */
+    public void setPriority(final int priority)
+    {
+        this.priority = priority;
+    }
+
+    /**
+     * location getter.
+     *
+     * @return the priority.
+     */
+    public ResourceLocation getLocation()
+    {
+        return location;
+    }
+
+    /**
+     * Setter for the location.
+     *
+     * @param location the new location attribute.
+     */
+    public void setLocation(final ResourceLocation location)
+    {
+        this.location = location;
+    }
+
+    /**
+     * attack getter.
+     *
+     * @return the attack.
+     */
+    public boolean getAttack()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename this method to start with "is" or "has". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2047)